src/hash/blake/mod.rs

```diff
@@ -0,0 +1,226 @@
+use crate::{
+    ByteReader, ByteWriter, Deserializable, DeserializationError, Digest, ElementHasher, Felt,
+    FieldElement, HashFn, Serializable, StarkField,
+};
+use core::{
+    mem::{size_of, transmute, transmute_copy},
+    ops::Deref,
+    slice::from_raw_parts,
+};
+
+#[cfg(test)]
+mod tests;
+
+// BLAKE3 N-BIT OUTPUT
+// ================================================================================================
+
+/// N-bytes output of a blake3 function.
+///
+/// Note: `N` can't be greater than `32`
```

Comment:
Can we elaborate it a bit more why it can't accept a `N` greater than `32`. I also wondered if we can put it under an error or panic.

Reply:
This is a restriction leaked from the upstream `Digest` definition.

We could implement support for 33+ bytes output, but since we don't have a variant for that, the runtime overhead isn't worthy.

I added a comment making reference to the Digest restriction